include/pisa/topk_queue.hpp

```diff
@@ -34,18 +51,21 @@ struct topk_queue {
         if (PISA_UNLIKELY(m_q.size() <= m_k)) {
             std::push_heap(m_q.begin(), m_q.end(), min_heap_order);
             if (PISA_UNLIKELY(m_q.size() == m_k)) {
-                m_threshold = m_q.front().first;
+                m_effective_threshold = m_q.front().first;
             }
         } else {
             std::pop_heap(m_q.begin(), m_q.end(), min_heap_order);
             m_q.pop_back();
-            m_threshold = m_q.front().first;
+            m_effective_threshold = m_q.front().first;
         }
         return true;
     }
 
-    bool would_enter(float score) const { return score >= m_threshold; }
+    /// Checks if an entry with the given score would be inserted to the queue, according
+    /// to the current threshold.
+    bool would_enter(float score) const { return score > m_effective_threshold; }
```

Comment:
I think this is fine depending on how `m_effective_threshold` is handled, see comments above...